core/Tracker/Db/Pdo/Mysql.php

```diff
@@ -209,6 +218,57 @@ public function fetch($query, $parameters = array())
      * @throws DbException if an exception occurred
      */
     public function query($query, $parameters = array())
+    {
+        try {
+            return $this->executeQuery($query, $parameters);
+        } catch (Exception $e) {
+            $isSelectQuery = stripos(trim($query), 'select ') === 0;
+            $isUpdateQuery = stripos(trim($query), 'update ') === 0;
+
+            if (($isSelectQuery || $isUpdateQuery)
+                && !$this->activeTransaction
+                && $this->isMysqlServerHasGoneAwayError($e)) {
+                // mysql may return a MySQL server has gone away error when trying to execute the query
+                // in that case we want to retry establishing the connection once after a short sleep
+                // we're only retrying SELECT and UPDATE queries to prevent inserting records twice for some reason
```

Comment:
Just a note: In theory an update statement could also result in incorrect values when executed twice. Like when increasing or decreasing by certain values.

Reply:
Good point @sgiehl I'll remove it just to be safe.